File: mtgSearch/src/main/java/com/dbottillo/view/MTGCardImageView.java
package com.dbottillo.view;

import android.content.Context;
import android.util.AttributeSet;
import android.widget.ImageView;

import com.dbottillo.cards.MTGCardFragment;

public class MTGCardImageView extends ImageView {
    public MTGCardImageView(Context context) {
        super(context);
    }

    public MTGCardImageView(Context context, AttributeSet attrs) {
        super(context, attrs);
    }

    public MTGCardImageView(Context context, AttributeSet attrs, int defStyleAttr) {
        super(context, attrs, defStyleAttr);
    }

    @Override
    public void onMeasure(int widthMeasureSpec, int heightMeasureSpec) {
        int height = MeasureSpec.getSize(heightMeasureSpec);
        setMeasuredDimension((int) (height / MTGCardFragment.RATIO_CARD), height);
    }
}

File: feature_base_cards/src/main/kotlin/com/dbottillo/mtgsearchfree/ui/views/MTGCardImageView.kt
package com.dbottillo.mtgsearchfree.ui.views

import android.content.Context
import androidx.appcompat.widget.AppCompatImageView
import android.util.AttributeSet
import com.dbottillo.mtgsearchfree.Constants.RATIO_CARD

class MTGCardImageView : AppCompatImageView {

    constructor(context: Context) : super(context)

    constructor(context: Context, attrs: AttributeSet) : super(context, attrs)

    constructor(context: Context, attrs: AttributeSet, defStyleAttr: Int) : super(context, attrs, defStyleAttr)

    override fun onMeasure(widthMeasureSpec: Int, heightMeasureSpec: Int) {
        super.onMeasure(widthMeasureSpec, heightMeasureSpec)
        val height = (measuredWidth * RATIO_CARD).toInt()
        setMeasuredDimension(measuredWidth, height)
    }
}